addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaZoneThingHandler.java

```diff
@@ -0,0 +1,621 @@
+            updateState(channelUID, new StringType(zoneState.surroundProgram));
+        } else if (id.equals(grpZone(YamahaReceiverBindingConstants.CHANNEL_VOLUME))) {
+            updateState(channelUID, new PercentType((int) zoneState.volume));
+        } else if (id.equals(grpZone(YamahaReceiverBindingConstants.CHANNEL_MUTE))) {
+            updateState(channelUID, zoneState.mute ? OnOffType.ON : OnOffType.OFF);
+
+        } else if (id.equals(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK))) {
+            updateState(channelUID, new StringType(playInfoState.playbackMode));
+
+        } else if (id.equals(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_STATION))) {
+            updateState(channelUID, new StringType(playInfoState.station));
+
+        } else if (id.equals(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_ARTIST))) {
+            updateState(channelUID, new StringType(playInfoState.artist));
+
+        } else if (id.equals(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_ALBUM))) {
+            updateState(channelUID, new StringType(playInfoState.album));
+
+        } else if (id.equals(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_SONG))) {
+            updateState(channelUID, new StringType(playInfoState.song));
+
+        } else if (id.equals(grpPlayback(YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_PRESET))) {
+            updateState(channelUID, new DecimalType(playControlState.presetChannel));
+
+        } else if (id.equals(grpNav(YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_MENU))) {
+            updateState(channelUID, new StringType(naviState.getCurrentItemName()));
+
+        } else if (id.equals(grpNav(YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_LEVEL))) {
+            updateState(channelUID, new DecimalType(naviState.menuLayer));
+
+        } else if (id.equals(grpNav(YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_CURRENT_ITEM))) {
+            updateState(channelUID, new DecimalType(naviState.currentLine));
+
+        } else if (id.equals(grpNav(YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_TOTAL_ITEMS))) {
+            updateState(channelUID, new DecimalType(naviState.maxLine));
+        } else {
+            logger.error("Channel not implemented: {}", id);
+        }
+    }
+
+    /**
+     * Call zoneControl.updateState() and will put the zone thing offline if there was a communication
+     * error. This is called by the bridge periodically.
+     */
+    public void updateZoneInformation() {
+        if (zone == null || zoneControl == null) {
+            return;
+        }
+        try {
+            zoneControl.updateState();
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        } catch (ParserConfigurationException e) {
+            logger.error(e.getLocalizedMessage());
```

Comment:
I might have missed something, [E.4 of the guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) describes log levels. Bindings should normally not log errors, please take a look at the page to check what would be proper here and adapt all `logger.error`'s